src/components/DragAndDrop/index.js

```diff
@@ -33,12 +36,18 @@ const defaultProps = {
         }
         return false;
     },
+    disabled: false,
 };
 
 export default class DragAndDrop extends React.Component {
     constructor(props) {
         super(props);
 
+        if (props.disabled) {
+            this.isDisabled = true;
```

Comment:
Why do this, vs just checking `this.props.disabled` in `componentDidMount` & `componentWillUnmount`?

Reply:
Ok that sounds understandable, but why don't we just remove all event listeners no matter what? There won't be an error removing event listeners that don't exist, right?

Reply:
> Even though disabled prop is set to true, listener callbacks are still triggered because already subscribed.

So wait in this case isn't the best solution to have `componentDidMount` check if `this.props.disabled` changes AND is now true (remove listener callbacks) or false (add listener callbacks), then we just always remove them in `componentWillUnmount`?

Just trying to get to the best solution here :D

Reply:
sorry which of those relates to the `componentDidMount` implementation?

Reply:
Arg I meant `componentDidUpdate` - in [this comment](https://github.com/Expensify/App/pull/13523#discussion_r1054535511) i meant we could check the prop value in `componentDidUpdate` sorry 🙃

Reply:
Latest changes look to add/remove listeners in `componentDidUpdate` when `props.disabled` changes. And `componentDidMount` and `componentWillUnmount` correctly checks `props.disabled` before adding/removing listeners. So I think @Beamanator's concerns should be all handled 👍🏽